src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
It does always reference FIE, but not until we make a Fake, and
TypeCatalogue loads first. Except in the integration tests. Or possibly
some other contrived example. Like if someone were to use Dynamic Proxy to
create a Dummy Factory. Which is unlikely, but would be a reason not to fix
this issue…